README.md

```diff
@@ -6,55 +7,297 @@ This project aims to build a non-conventional alarm clock with smart features su
 
 [GSoC page](https://summerofcode.withgoogle.com/programs/2023/projects/c7GIl1mf)
 
-## Getting Started
+## Table of Contents
+- [GetX Pattern](#getx-pattern)
+- [Database Schema](#database-schema)
+- [Installation & Setup](#installation-setup)
+- [User Interface & Features](#user-interface-features)
+- [Contribution Guidelines](#contribution-guidelines)
+- [Future Plans](#future-plans)
+- [Community](#community)
+- [Flutter](#flutter)
+
+## GetX Pattern
```

Comment:
We should also mention about using get_cli to create new pages to make everything easier

Also, we need to briefly touch upon what each of those files do in general (controller, view etc).

Reply:
Okay, I'll be adding this.